src/idpyoidc/client/service.py

```diff
@@ -628,27 +634,42 @@ def parse_response(
         LOGGER.debug("response_cls: %s", self.response_cls.__name__)
 
         if resp is None:
-            if not info:
+            if self.response_cls == list and info == []:
+                return info
+            elif not info:
                 LOGGER.error("Missing or faulty response")
                 raise ResponseError("Missing or faulty response")
 
             if sformat == "text":
                 resp = info
             else:
                 resp = self._do_response(info, sformat, **kwargs)
-                LOGGER.debug('Initial response parsing => "%s"', resp.to_dict())
+                if isinstance(resp, Message):
+                    LOGGER.debug(f'Initial response parsing => "{resp.to_dict()}"')
+                else:
+                    LOGGER.debug(f'Initial response parsing => "{resp}"')
 
         # is this an error message
         if sformat == "text":
             pass
         elif is_error_message(resp):
             LOGGER.debug("Error response: %s", resp)
-        else:
+        elif isinstance(resp, Message):
             vargs = self.gather_verify_arguments(response=resp, behaviour_args=behaviour_args)
             LOGGER.debug("Verify response with %s", vargs)
             try:
                 # verify the message. If something is wrong an exception is thrown
                 resp.verify(**vargs)
+            except MissingSigningKey as err:
+                LOGGER.error(f"Could not find an appropriate key: {err}")
+                _keyjar = self.upstream_get("attribute", "keyjar")
+                try:
```

Comment:
probably there are ways to avoid nested try/except here

Reply:
Simplified the code.